### PR TITLE
feat: support Pydantic V2

### PR DESCRIPTION
**Pull request by AtomHare**
### Description

This PR adds the necessary modifications in order to support Pydantic breaking changes !

HOWEVER, do not use in production already, please test it thoroughly to verify that everything is working.

### CODE REVIEW

**Review by armanddidierjean — CHANGES_REQUESTED** [review]
Creating new loan items does not work with Titan:

 - on creation Hyperion send a 201 but Titan raise an error
 - on getting items Titan raise an error "get a string instead of float"

**Comment by armanddidierjean** [comment]
Do you think you could write a migration file to change the Interval to an Int in the database @Rotheem?
Capture d’écran 2024-01-14 à 19 55 03

File: app/schemas/schemas_loan.py [hunk 1]
```diff
@@ -30,17 +28,17 @@ class ItemBase(BaseModel):
...
+    suggested_lending_duration: int = Field(description="duration in seconds")
+    model_config = ConfigDict(from_attributes=True)
 
 
 class ItemUpdate(BaseModel):
     name: str | None = None
     suggested_caution: int | None = None
     total_quantity: int | None = None
-    suggested_lending_duration: timedelta | None = None
+    suggested_lending_duration: int | None = Field(
```
[the remark this answers is not in the document]
Comment (armanddidierjean) [reply app/schemas/schemas_loan.py:39]: This should be two equivalent syntaxes

File: app/schemas/schemas_notification.py [hunk 2]
```diff
@@ -15,26 +15,23 @@ class Message(BaseModel):
...
     expire_on: date
-
-    class Config:
-        orm_mode = True
+    model_config = ConfigDict(from_attributes=True)
 
 
 class FirebaseDevice(BaseModel):
     user_id: str = Field(description="The Hyperion user id")
     firebase_device_token: str = Field("Firebase device token")
```
[the remark this answers is not in the document]
Comment (AtomHare) [reply app/schemas/schemas_notification.py:36]: You're right this should be changed however it's not in the scope of this PR

File: app/endpoints/amap.py [hunk 3]
```diff
@@ -70,7 +70,7 @@ async def create_product(
 
     **The user must be a member of the group AMAP to use this endpoint**
     """
-    db_product = models_amap.Product(id=str(uuid.uuid4()), **product.dict())
+    db_product = models_amap.Product(id=str(uuid.uuid4()), **product.__dict__)
```
[the remark this answers is not in the document]
Comment (AtomHare) [reply app/endpoints/amap.py:73]: If I recall correctly  `model_dump` doesn't work here

**Review by guillaumetavernier — APPROVED** [review]

**Review by Rotheem — APPROVED** [review]

**Review by armanddidierjean — APPROVED** [review]
I couldn't test the PR, but the code looks good to me

**Pull request merged**